File: src/fiefapp.svelte/src/models/settings/livingcondition-alternative.ts
import { Livingcondition, LivingconditionType } from '../livingcondition';

interface ILivingconditionAlternative {
  type: string;
  baseCost: number;
  luxuryCost: number;
  focusGain: number;
  wellbeingGain: number;
  description: string;
}

export class LivingconditionAlternative implements ILivingconditionAlternative {
  constructor(
    public type: string,
    public baseCost: number,
    public luxuryCost: number,
    public focusGain: number,
    public wellbeingGain: number,
    public description: string
  ) {}

  public create(): Livingcondition {
    return new Livingcondition(
      LivingconditionType[this.type as keyof typeof LivingconditionType],
      this.baseCost,
      this.luxuryCost,
      this.focusGain,
      this.wellbeingGain,
      this.description
    );
  }
}

File: src/fiefapp.entities/LivingconditionAlternative.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace fiefapp.entities
{
    public class LivingconditionAlternative
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        public int BaseCost { get; set; }
        public int LuxuryCost { get; set; }
        public int FocusGain { get; set; }
        public int WellbeingGain { get; set; }
        public string Description { get; set; }
    }
}
